web/vibe/web/rest.d

```diff
@@ -1859,47 +1948,46 @@ private Json request(URL base_url,
 			req.writeBody(cast(const(ubyte)[])body_, hdrs.get("Content-Type", "application/json"));
 	};
 
-	auto resdg = (scope HTTPClientResponse res) {
-		if (!res.bodyReader.empty)
-			ret = res.readJson();
-
-		logDebug(
-			 "REST call: %s %s -> %d, %s",
-			 httpMethodString(verb),
-			 url.toString(),
-			 res.statusCode,
-			 ret.toString()
-			 );
-
-		// Get required headers - Don't throw yet
-		string[] missingKeys;
-		foreach (k, ref v; reqReturnHdrs.byKeyValue)
-			if (auto ptr = k in res.headers)
-				v = (*ptr).idup;
-			else
-				missingKeys ~= k;
-
-		// Get optional headers
-		foreach (k, ref v; optReturnHdrs.byKeyValue)
-			if (auto ptr = k in res.headers)
-				v = (*ptr).idup;
-			else
-				v = null;
-
-		if (missingKeys.length)
-			throw new Exception(
-				"REST interface mismatch: Missing required header field(s): "
-				~ missingKeys.to!string);
-
-
-		if (!isSuccessCode(cast(HTTPStatus)res.statusCode))
-			throw new RestException(res.statusCode, ret);
-	};
-
-	if (http_settings) requestHTTP(url, reqdg, resdg, http_settings);
-	else requestHTTP(url, reqdg, resdg);
+	HTTPClientResponse client_res;
+	if (http_settings) client_res = requestHTTP(url, reqdg, http_settings);
+	else client_res = requestHTTP(url, reqdg);
+
+	import vibe.stream.operations;
+
+	logDebug(
+			"REST call: %s %s -> %d, %s",
+			httpMethodString(verb),
+			url.toString(),
+			client_res.statusCode,
+			ret
+			);
+
+	// Get required headers - Don't throw yet
+	string[] missingKeys;
+	foreach (k, ref v; reqReturnHdrs.byKeyValue)
+		if (auto ptr = k in client_res.headers)
+			v = (*ptr).idup;
+		else
+			missingKeys ~= k;
+
+	// Get optional headers
+	foreach (k, ref v; optReturnHdrs.byKeyValue)
+		if (auto ptr = k in client_res.headers)
+			v = (*ptr).idup;
+		else
+			v = null;
+	if (missingKeys.length)
+		throw new Exception(
+			"REST interface mismatch: Missing required header field(s): "
+			~ missingKeys.to!string);
```

Comment:
Note for the future: I'd really like us to improve this part so it doesn't allocate this much...